src/main/scala/com/gu/datalakealerts/SchedulerLambda.scala

```diff
@@ -18,31 +22,44 @@ object SchedulerLambda {
 
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
-  def handler(lambdaInput: LambdaInput, context: Context): Unit = {
+  def handler(context: Context): Unit = {
     val env = SchedulerEnv()
     logger.info(s"Starting $env")
     process(env)
   }
 
-  def process(env: SchedulerEnv, dryRun: Boolean = false): Unit = {
+  def startQuery(env: SchedulerEnv, event: MonitoringEvent): Try[String] = Try {
+    logger.info(s"Starting monitoring for ${event.feature} on ${event.platform}")
+    val monitoringQuery = event.feature.monitoringQuery(event.platform)
+    logger.info(s"Query will be:\n ${monitoringQuery.query}")
+    Athena.startQuery(monitoringQuery).getQueryExecutionId
+  }
+
+  def queueQuery(env: SchedulerEnv, event: MonitoringEvent, queryExecutionId: String): Try[SendMessageResult] = Try {
+    val eventWithQueryInfo = MonitoringEventWithQueryInfo(event, queryExecutionId)
+    val queueName = Sqs.queueName(env.app, env.stage)
+    logger.info(s"Adding query id $queryExecutionId to the $queueName queue so that the results can be checked later...")
+    Sqs.enqueueRunningQuery(eventWithQueryInfo, queueName, 30)
```

Comment:
Are you delaying here to ensure AWS has updated its state about the running queries?
If so I'd recommend adding a tiny comment explaining why

Reply:
Actually, since most queries take several seconds to complete anyway, the main goal here was to avoid unnecessary polling. I'll add a comment to explain this.